### DISCUSSION TITLE
LifeFreeze and PowerFreeze flags for AssertSpecial

### DISCUSSION THREAD

**Discussion by UrsoManso** [ikemen-engine/Ikemen-GO#1330] [Feature Request] [answered]
### Additional context

I thought it would be nice to have those to make coding effects that would freeze your life and/or power during specified conditions a lot more convinient.

**Comment by RealFoobs** [comment]
Don't think assertspecial flags would be the way to go here. 

Something that has been floated around on the MFG server is a "powermulset" state controller; powermulset = 0 would functionally freeze the character's power gain, but the state controller has more uses, such as coding a power gain "decay" system during combos, akin to damage dampening. This is possible with current tools, but having a state controller would be much easier than needing to rig vars/maps and adding conditionals to every hitdef.

**Reply by UrsoManso** [reply]
Sounds good. But the reason I've suggested flags is because in my head it seems easier to implement and therefore higher the chances of actually being implemented. 😄

**Comment by potsmugen** [comment, marked as answer]
Those could be useful but you can already freeze those values on your char or anyone else with RedirectID. There's also flags to disable damage already.

**Reply by UrsoManso** [reply]
How exactly can RedirectID be used to freeze powerbar?

**Reply by potsmugen** [reply]
You store the player's current power in a map then powerset them continuously to that value.

**Reply by UrsoManso** [reply]
Can this be achieved under negative states?
